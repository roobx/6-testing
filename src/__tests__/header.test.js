import { cleanup, render, screen } from "@testing-library/react";
import { Provider } from "react-redux";
import Header from "../components/Header";
import store from "../store/store";
import { Router } from "react-router-dom";
import { createMemoryHistory } from "history";
import userEvent from "@testing-library/user-event";

describe("Header component", () => {
    afterEach(cleanup);

    it("Should render logo", () => {
        const altText = "Логотип проекта Mesto";
        const headerWrapperId = "header-wrapper-element";
        const history = createMemoryHistory();
        render(
            <Router history={history}>
                <Provider store={store}>
                    <Header />
                </Provider>
            </Router>
        );

        const img = screen.getByAltText(altText);
        expect(img).toHaveClass("logo");

        const loginWrapper = screen.getByTestId(headerWrapperId);
        expect(loginWrapper).toHaveClass("header__wrapper");
    });

    it("Should render logo", async () => {
        const history = createMemoryHistory();

        history.push("/signin");

        render(
            <Router history={history}>
                <Provider store={store}>
                    <Header />
                </Provider>
            </Router>
        );

        expect(screen.getByTestId("link-signup-element")).toHaveTextContent("Регистрация");
        expect(history.location.pathname).not.toEqual("/login");
        expect(history.location.pathname).toEqual("/signin");

        await userEvent.click(screen.getByTestId("link-signup-element"));

        expect(screen.getByTestId("link-signin-element")).toHaveTextContent("Войти");
        expect(history.location.pathname).toEqual("/signup");
    });

    it("Should render logo", async () => {
        const history = createMemoryHistory();

        const { asFragment } = render(
            <Router history={history}>
                <Provider store={store}>
                    <Header />
                </Provider>
            </Router>
        );

        expect(asFragment()).toMatchSnapshot();
    });
});
