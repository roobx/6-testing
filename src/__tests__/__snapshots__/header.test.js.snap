// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component Should render logo 1`] = `
<DocumentFragment>
  <header
    class="header page__section 
                 
                header_page-main"
  >
    <a
      class="header__auth-link"
      href="/"
    >
      <img
        alt="Логотип проекта Mesto"
        class="logo header__logo"
        src="logo.svg"
      />
    </a>
    <button
      aria-label="меню"
      class="header__burger"
      type="button"
    />
    <div
      class="header__wrapper"
      data-testid="header-wrapper-element"
    >
      <p
        class="header__user"
      />
      <button
        class="header__logout"
      >
        Выйти
      </button>
    </div>
  </header>
</DocumentFragment>
`;
